vispy/scene/widgets/widget.py

```diff
@@ -63,9 +63,13 @@ def __init__(self, pos=(0, 0), size=(10, 10), border_color=None,
         self._size = 100, 100
 
         # layout interaction
-        # todo: use Cassowary; see #277
-        self._fixed_size = (None, None)
-        self._stretch = (None, None)
+        self._width_limits = (0, None)
+        self._height_limits = (0, None)
+        self._stretch = (1, 1)
```

Comment:
you can make `stretch` a `kwarg`, that way when the `Widget` is created by the grid system it can set it appropriately based on the stretch